src/status_im/contexts/profile/contact/share/view.cljs

```diff
@@ -0,0 +1,52 @@
+(ns status-im.contexts.profile.contact.share.view
+  (:require [legacy.status-im.ui.components.list-selection :as list-selection]
+            [quo.core :as quo]
+            [react-native.core :as rn]
+            [react-native.safe-area :as safe-area]
+            [status-im.common.qr-codes.view :as qr-codes]
+            [status-im.contexts.profile.contact.share.style :as style]
+            [status-im.contexts.profile.utils :as profile.utils]
+            [utils.address :as address]
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]))
+
+(defn view
+  []
+  (let [{:keys [universal-profile-url customization-color]
+         :as   profile} (rf/sub [:contacts/current-contact])
+        abbreviated-url (rn/use-memo (fn []
+                                       (address/get-abbreviated-profile-url universal-profile-url))
+                                     [universal-profile-url])
+        on-back-press   #(rf/dispatch [:navigate-back])
```

Comment:
I think we might want to extract this into a external `defn` if we want to avoid passing anonymous function refs as props. Thoughts?

Reply:
I think i moved it here because i wanted to use use-callback but not that i removed it maybe i can still use func here again, wdyt? @seanstrom 

```
:on-press            #(rf/dispatch [:navigate-back])
```

Reply:
I think it sorta depends on what's receiving `on-press`, because passing the anonymous function `#(rf/dispatch [:navigate-back])` can cause the receiving component to re-render. I think this is because the anonymous function will be re-computed each time the contact-share-view component re-renders.

In this case, I think the receiving component is `page-top`, and we can decide if we need to avoid possibly re-rendering that component. If we do and we don't need `use-callback`, then we can use `defn` as a static function and avoid the possible extra re-renders.

Reply:
thank you for explanation, make sense. just updated

Reply:
yes, I believe @seanstrom is correct, in this case since it doesn't depend on any parameter, it should be outside so that even if it's compared by reference, it will hold equal and not cause unnecessary re-rendering, one less piece of state to worry about